kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/strimzi/Strimzi.java

```diff
@@ -35,7 +48,40 @@ public class Strimzi {
      */
     public Strimzi(String deploymentNamespace) throws IOException {
         this.deploymentNamespace = deploymentNamespace;
-        deployment = kubeClient().getClient().load(DeploymentUtils.getDeploymentFileFromURL(Environment.STRIMZI_URL));
+        InputStream strimziYaml = DeploymentUtils.getDeploymentFileFromURL(Environment.STRIMZI_URL);
+        InputStream strimziYamlKRaft = configureKRaftModeForStrimzi(strimziYaml);
+        deployment = kubeClient().getClient().load(strimziYamlKRaft);
+    }
+
+    private InputStream configureKRaftModeForStrimzi(InputStream strimziYaml) throws IOException {
+        YAMLFactory yamlFactory = new YAMLFactory();
+        ObjectMapper mapper = new YAMLMapper();
+
+        YAMLParser yamlParser = yamlFactory.createParser(strimziYaml);
+        List<JsonNode> docs = mapper
+                .readValues(yamlParser, new TypeReference<JsonNode>() {
+                })
+                .readAll();
+        boolean found = false;
+        for (JsonNode doc : docs) {
+            String kind = doc.at("/kind").asText("");
+            if (kind.equals(Constants.DEPLOYMENT)) {
+                ArrayNode arrayNode = (ArrayNode) doc.at("/spec/template/spec/containers");
+                ArrayNode envNode = (ArrayNode) arrayNode.get(0).at("/env");
+                for (JsonNode node : envNode) {
+                    if (node.at("/name").asText().equals(Environment.STRIMZI_FEATURE_GATES_ENV)) {
+                        found = true;
+                        ((ObjectNode) node).put("value", String.join(",", Constants.USE_KRAFT_MODE, Constants.USE_KAFKA_NODE_POOLS));
```

Comment:
it'd have been slightly nicer to have append to whatever was already present, but we aren't going to need this code for too long.

Reply:
Right, but modified just in case.